draft-ietf-tls-esni.md

```diff
@@ -761,6 +774,21 @@ If the server sends a NewSessionTicket message, the corresponding ECH PSK MUST
 be ignored by all other servers in the deployment when not negotiating ECH,
 including servers which do not implement this specification.
 
+## Backend Server Behavior {#backend-server-behavior}
+
+When the client-facing server accepts ECH, it forwards the ClientHelloInner to
+the backend server, who terminates the connection. If the ClientHelloInner
+contains an empty "encrypted_client_hello" extension, then the backend server
+MUST confirm ECH acceptance by setting ServerHello.random[24:32] to
+
+~~~~
+    accept_confirmation = HKDF-Extract(0,
+        ClientHelloInner.random + ServerHello.random[0:24])[0:8]
```

Comment:
Did we decide that HKDF-Extract was OK? We never use raw HKDF-Extract anywhere else in TLS 1.3.

Reply:
It seems fine to me, since the ClientHelloInner.random is secret and we need is a pseudorandom string. I'll take another look at the HKDF paper and get back to you. (Also, it would be great to hear from others on this.)

Reply:
Judging from [1, Section 6], just using HKDF-Extract might be fine, but it seems rather "hacky". I think we should err on the conservative side and use HKDF-Extract-then-Expand. Will update the PR to reflect this.